### PR TITLE
Update EP_AGORA configuration

### PR DESCRIPTION
**Pull request by trowbridgec**
Rearrange `EP_AGORA` configuration to move KVSTORE to the end of flash to fix bug with Pelion OTA updates

### CODE REVIEW

**Comment by teetak01** [comment]
What is the OTA issue this is resolving ?

**Comment by trowbridgec** [comment]
The client doesn't seem to like having the KVSTORE right after the bootloader in internal flash (see layout below). With this config, Pelion updates would fail to start and report an error of 1037 (`UpdateWarningUnknown`).
```
+--------------------------+
|                          |
|                          |
|                          |
|        Active App        |
|                          |
|                          |
|                          |
+--------------------------+
|Active App Metadata Header|
+--------------------------+
|                          |
|         KVSTORE          |
|                          |
+--------------------------+
|                          |
|        Bootloader        |
|                          |
+--------------------------+
```
 I was never able to track down the initial source of the failure, but switching up the internal flash configuration to move the KVSTORE to the end of flash seems to have fixed the issue:
```
+--------------------------+
|                          |
|         KVSTORE          |
|                          |
+--------------------------+
|                          |
|                          |
|                          |
|        Active App        |
|                          |
|                          |
|                          |
+--------------------------+
|Active App Metadata Header|
+--------------------------+
|                          |
|        Bootloader        |
|                          |
+--------------------------+
```

File: configs/external_kvstore_with_spif.json [hunk 1]
```diff
@@ -33,11 +34,15 @@
...
-                    "3. Active App Metadata Header - (1KiB/2KiB) from the end of KVSTORE",
-                    "4. Active App - From end of header to the end of flash",
+                    "2. Active App Metadata Header - (1KiB/2KiB) from the end of the bootloader",
+                    "3. Active App - From end of header to KVSTORE",
+                    "4. KVSTORE - 2 flash sectors at the end of flash",
                     "+--------------------------+",
                     "|                          |",
+                    "|         KVSTORE          |",
+                    "|                          |",
+                    "+--------------------------+ <-+ storage_tdb_internal.internal_base_address",
```
Comment (teetak01) [inline configs/external_kvstore_with_spif.json:44]: This comment is not correct, as it refers to TDB_INTERNAL configuration, while this file is using FILESYSTEM.

**Comment by teetak01** [comment]
I think that is more likely due to empty/dummy `update_default_resources.c` or update-image that has not been compiled with compatible `update_default_resources.c`

Only functional difference here is that during flashing KVstore (internal part) is erased if it is located between bootloader and application (bin-flashing, hex-flashing would also preserve that). But because the application is doing hardcoded entropy injection in startup it is still able to reopen same developer credentials.

This is most common configuration currently, so unless this is something target specific, I do not think this should be needed. To me it would seem that the incompatibility case would happen equally with this or previous implementation.

**Comment by loverdeg** [comment]
All for bringing KVStore internal. But also agree, root cause should be identified.

**Comment by trowbridgec** [comment]
@teetak01 and @loverdeg: I'll see if I can re-create the issue with the current config and compare it to the proposed changes. I don't know for sure until I verify, but I think it's something other than the `update_default_resources.c` file; my guess was that the partitions of the flash (bootloader, KVSTORE, etc.) didn't cleanly line up with the flash regions of the part? Just a guess...

**Comment by trowbridgec** [comment]
Well, looks like I was wrong...

I rebuilt with the current config (`master` of `mbed-os-example-pelion`) and updates work just fine. It must have been that I was using a bad `update_default_resources.c` file.

@teetak01 What would you like to do with this PR? Leave the `EP_AGORA` config as is and close this? We'd probably also need to close the companion PR I opened for the `mbed-os-example-pelion app` as well (https://github.com/ARMmbed/mbed-os-example-pelion/pull/143).

**Comment by teetak01** [comment]
Well, its up to you mostly, but I think there is no real reason to start changing the configurations now.

**Comment by 0Grit** [comment]
My recommendation would be to close and start a new PR to get those secrets on-die.
I would however look into how Nordic ended up getting this part PSA certified without it having a true secure boot implementation. The answer may be a factor in this decision.
.

**Comment by teetak01** [comment]
If you can just fit your use-case on the internal flash, its much easier and more efficient. You save both storage drivers and filesystem drivers there.
Or even just switching from FILESYSTEM => TDB_EXTERNAL would already save the filesystem drivers from RAM and ROM.

With presumption that other parts of the usecase does not depend on external storage and/or filesystem.

**Comment by teetak01** [comment]
Closing based on discussion. Can be reopened if there is need.

**Pull request closed without merging**